src/main/java/org/dswarm/xsd2jsonschema/JsonSchemaParser.java

```diff
@@ -54,155 +57,216 @@
 
 public class JsonSchemaParser {
 
-	private final XSOMParser parser = new XSOMParser();
+	private static final String HASH        = "#";
+	private static final String HTTP_PREFIX = "http://";
+	private static final String SLASH       = "/";
+	private static final String AT          = "@";
+	private static final String WILDCARD    = "wildcard";
+	private static final String NULL        = "null";
+
+	private static final XSImplementation impl         = (XSImplementation) new XSImplementationImpl();
+	private static final XSLoader         schemaLoader = impl.createXSLoader(null);
+
+	private XSModel model = null;
 
 	private List<JSElement> iterateParticle(final XSParticle particle) {
 
 		final XSTerm term = particle.getTerm();
 
-		if (term.isModelGroup()) {
+		if (term instanceof XSModelGroup) {
 
-			final XSModelGroup modelGroup = term.asModelGroup();
+			final XSModelGroup modelGroup = (XSModelGroup) term;
 			return iterateModelGroup(modelGroup);
-		} else if (term.isModelGroupDecl()) {
+		}
+
+		// TODO: dunno how to replace this? -> we can cast it to XSModelGroup but maybe it should ba a XSModelGroupDefinition and has something to do with substition groups
+		/*else if (term.isModelGroupDecl()) {
 
 			final XSModelGroupDecl xsModelGroupDecl = term.asModelGroupDecl();
 			return iterateModelGroup(xsModelGroupDecl.getModelGroup());
 		}
+		*/
 
 		final ArrayList<JSElement> jsElements = new ArrayList<>(1);
-		jsElements.add(iterateSingleParticle(particle));
+		final Optional<JSElement> optionalJSElement = iterateSingleParticle(particle);
+
+		if (optionalJSElement.isPresent()) {
+
+			jsElements.add(optionalJSElement.get());
+		}
```

Comment:
Could be `jsElements.addAll(optionalJSElement.toSet())` (or similar), which remove the isPresent/get checks.